rasdaemon.c

```diff
@@ -74,13 +119,38 @@ int main(int argc, char *argv[])
 {
 	struct arguments args;
 	int idx = -1;
+
+	const struct argp_option offline_options[] = {
+		{"smca", SMCA, 0, 0, "AMD SMCA Error Decoding"},
+		{"model", MODEL, 0, 0, "CPU Model"},
+		{"family", FAMILY, 0, 0, "CPU Family"},
+		{"bank", BANK_NUM, 0, 0, "Bank Number"},
+		{"ipid", IPID_REG, 0, 0, "IPID Register (for SMCA systems only)"},
+		{"status", STATUS_REG, 0, 0, "Status Register"},
+		{0, 0, 0, 0, 0, 0},
+	};
+
+	struct argp offline_argp = {
+		.options = offline_options,
+		.parser = parse_opt_offline,
+		.doc = TOOL_DESCRIPTION,
+		.args_doc = ARGS_DOC,
+	};
+
+	struct argp_child offline_parser[] = {
+		{&offline_argp, 0, "Post-Processing Options:", 0},
+		{0, 0, 0, 0},
+	};
+
 	const struct argp_option options[] = {
 		{"enable",  'e', 0, 0, "enable RAS events and exit", 0},
 		{"disable", 'd', 0, 0, "disable RAS events and exit", 0},
 #ifdef HAVE_SQLITE3
 		{"record",  'r', 0, 0, "record events via sqlite3", 0},
 #endif
 		{"foreground", 'f', 0, 0, "run foreground, not daemonize"},
+		{"post-processing", 'p', 0, 0,
+		"Post-processing MCE's with raw register values"},
```

Comment:
These need to be around #ifdef too

Reply:
Need to make sure MCE specific options won't show on a binary compiled without MCE

Reply:
Will handle it but this might require ifdef's in a few other places as well to avoid compiler warnings for unused variables. Something like below (only snippets attached):


@@ -62,11 +74,48 @@ static error_t parse_opt(int k, char *arg, struct argp_state *state)
        case 'f':
                args->foreground++;
                break;
+#ifdef HAVE_MCE
+       case 'p':
.......
+               break;
+#endif
+       default:
=========================================================

+#ifdef HAVE_MCE
+static error_t parse_opt_offline(int key, char *arg,
+                                struct argp_state *state)
+{
......
        }
        return 0;
 }
+#endif
 =======================================================

@@ -74,6 +123,31 @@ int main(int argc, char *argv[])
 {
        struct arguments args;
        int idx = -1;
+
+#ifdef HAVE_MCE
+       const struct argp_option offline_options[] = {
.....
+       };
+
+       struct argp offline_argp = {
.....
+       };
+
+       struct argp_child offline_parser[] = {
.....
+       };
+#endif
===============================================

                {"foreground", 'f', 0, 0, "run foreground, not daemonize"},
+#ifdef HAVE_MCE
+               {"post-processing", 'p', 0, 0,
+               "Post-processing MCE's with raw register values"},
+#endif
 
                { 0, 0, 0, 0, 0, 0 }
==================================================

@@ -89,7 +167,9 @@ int main(int argc, char *argv[])
                .parser = parse_opt,
                .doc = TOOL_DESCRIPTION,
                .args_doc = ARGS_DOC,
-
+#ifdef HAVE_MCE
+               .children = offline_parser,
+#endif
        };

Sounds good?

Reply:
Yes, that'd work

Reply:
Okay... thank you for your assistance!
Will update thee pull request